data/src/test/java/org/apache/iceberg/data/GenericAppenderHelper.java

```diff
@@ -43,18 +45,28 @@ public class GenericAppenderHelper {
 
   private final Table table;
   private final FileFormat fileFormat;
-  private final TemporaryFolder tmp;
+  private final Path tmp;
   private final Configuration conf;
 
+  @Deprecated()
   public GenericAppenderHelper(
-      Table table, FileFormat fileFormat, TemporaryFolder tmp, Configuration conf) {
+      Table table, FileFormat fileFormat, TemporaryFolder tmp, Configuration conf)
```

Comment:
let's roll back these changes and focus only on adding the `ParameterizedTestExtension` and converting `TestDictRowGroupFilter´